### ISSUE TITLE
Aspire Story for User Secrets

### ISSUE THREAD

**Issue by ElanHasson** [dotnet/runtime#107849]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

_Note: I initially thought this could fit well into aspire, but as I was typing this up, I realized it may be a better fit for `dotnet` proper. Feel free to move this issue there._

I wish there was a better story around User Secrets for teams or even just new machines.

I got a new computer, cloned my repo, started my project and got an `OptionsVaidator` exception and it took me a few min to figure out _what_ was missing. 

I've invested some time setting up Aspire just right and the project, so it didn't rely on external values for things-- easy to get started day one for new team members: clone repo, F5.

Then I realized it was my User Secrets that were missing-- they're pretty much set and forget-- and they stay on the local machine for obvious reasons.

So, new machine, I have to set them. One more thing to remember to do. One more document to write. One more document to keep in sync. 😢 

### Describe the solution you'd like

I'd love to see a way to provide a default `secrets.json` for an Aspire project (or even a `csproj` as this would benefit non-aspire projects as well).

I do not think we need a new mechanism for this as I _think_  this would fit well into the `csproj` and this could probably implemented via a MSBuild task.

 ```xml
<Project Sdk="Microsoft.NET.Sdk.Web">
    <PropertyGroup>
		<!--Exists Today-->
		<UserSecretsId>37913f03-ffcc-4c77-9904-6b7241929494</UserSecretsId>
		<!--Proposed-->
		<UserSecrets>
			{
			"clientId": "",
			"clientSecret": "" // Get creds from https://my-intranet/provision-client,
                        "secret_tokens": []
                        "complexSecrets": {
                           "userId": 43110,
			}
		</UserSecrets>
	</PropertyGroup>
</Project>
```

## How would it work?

The point here is not to require a schema be specified, but only structural validation. The assumption is all values are required, values and types aren't validated, the exception is for properties that are object types: we would validate the path exists (`$.complexSecrets.userId` using above sample). 

### Non-Aspire

When you open a project in VS or Code, the host would see the `UserSecrets` element and see the current user secrets specified by `UserSecretsId` do not match the template specified in the `csproj` and alert the user that they are required and show them the template.

When you use the CLI, validation would occur at build time and the problem would be surfaced there.

### Aspire 

I'm not sure what this would look like in aspire, perhaps the above Non-Aspire workflow is better suited to be implemented as a `UserSecrets` resource that performs the validation. 

I think it maybe makes sense to do so as Aspire is responsible for downstream config injection to the projects and user secrets do get injected via a different pipeline outside of Aspire.

### Additional context

Related Issues: https://github.com/dotnet/aspire/issues/4035 https://github.com/dotnet/aspire/pull/2953#issuecomment-2004259054

**Comment by Clockwork-Muse** [comment]
If these are shared/well-known "local development only" secrets (eg, the password for the Cosmos DB Emulator), you can just put them into the standard development-only properties file, `<ProjectFolder>/Properties/launchSettings.json`, and just persist them in the repo.  These properties are hydrated into the normal config system on debug launch, but aren't published when the application is packaged.

For accessing a shared (probably cloud) resource, in many cases tooling is able to source login credentials from the local user profile (eg, from `~/.azure` for those APIs).  Written code doesn't explicitly have configuration for a password or other secret, but automatically grabs the user token from there, which is way more ergonomic since it allows for "just using" what would be the managed identity token when deployed (no development vs deploy code changes for user code)

**Comment by julealgon** [comment]
While I understand where the author is coming from here, I don't think this is a good solution, if only because of how general purpose the `IConfiguration` system is and the fact that user secrets is just another source for that.

If the problem is that "I lost settings when moving from computer A to computer B", I'd rather have a solution where the secrets were _not_ sourced from the machine, but instead from, say, Azure Key Vault.

It would be nice if it was possible for user secrets to be stored in an instance of KeyVault on Azure and tied to the user's account in Visual Studio, so that they would be downloaded seamlessly as needed and would persist across machines, using the same GUIDs that are used today to identify the projects.

**Comment by thompson-tomo** [comment]
I agree with @julealgon i don't like the proposed approach of polluting the csproj file and much more in favour of the key vault approach.

As such what I would like to see is the introduction of `Microsoft.Extensions.Vault` that offers `ISecretStore` which behaves similar to `Microsoft.Extensions.Configuration` but tailored for handling secrets. 

What I mean by this is:
- Providers/Stores can be added ie Azure Key Vault, Json
- A secret can be obtained from the vault
- A secret can be added/updated in the vault

Q's
- Should we include a freshness value for secrets?

**Comment by ElanHasson** [comment]
👋🏼, OP here.

I do agree,  the proposed solution isn't the best.

I was trying to express the need for a "user secrets schema" that tells you what is needed. That's a key pain point right there. Currently, you have to guess and read lots of code to figure out what's needed, what format, etc.


As for storage, 
I understand the desire to have a Azure key vault provider, but I feel like a secret gist provider could also be useful to non-cloud people.


Maybe it's as simple as having `dotnet user-secretd sync gist.github.com/ElanHasson/717481984691863719737194717` can use `gh-cli` under the hood to auth.

**Comment by thompson-tomo** [comment]
The same pain point can exist for configuration depending on your implementation.

With my approach for the vault which resembles configuration you could have it throw an exception if the json store can not be found when not in production. We could even have the launchsettings.json as an optional store hence greatest flexibility. Key thing being multiple stores which could even include github
